lib/lhm/chunker.rb

```diff
@@ -27,7 +27,7 @@ def initialize(migration, connection = nil, options = {})
     def execute
       return unless @start && @limit
       @next_to_insert = @start
-      while @next_to_insert < @limit || (@next_to_insert == 1 && @start == 1)
+      while @next_to_insert < @limit || (@start == @limit)
         stride = @throttler.stride
```

Comment:
The values @start and @limit are calulated from the table, so in this case its assume the row has id of 42..
start will be 42
limit will be 42

after line 29, the values are
next_to_insert = 42
start = 42
limit = 42

Note in the while loop the only thing that changes is the next_to_insert

so (@start == @limit) would cause the test to be true.. always.. never stopping